syntax = "proto3";
package protocol;

enum DataType
{
    HeartBeat = 0;
    reqcheckdir = 1;
    FileDataMsg = 2;
}

message Heart
{
	bool TCPServerStatus = 1;
	bool TCPClientStatus = 2;
}

message File
{
    string filename = 1;
    int32 size = 2;
    bytes relativepath = 3;
    int32 crcvalue = 4;
}

message Directory
{
    repeated Directory DirList = 1;
    repeated File FileList = 2;
    bytes relativepath = 3;
}

message DataInfo
{
    int32 msgid = 1;
    bytes info = 2;
}